### PR TITLE
Add api/topics endpoint with tests

### PR DESCRIPTION
**Pull request by solveigag**
Tested for array of objects, expected length and keys. For errors, just if the path is incorrect.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            expect(allTopics).toBeInstanceOf(Array)
+            expect(allTopics[0]).toBeInstanceOf(Object);
+            expect(allTopics).toHaveLength(3);
+          })
+    })
+    test("endpoint responds with an array of topic objects, which include description and slug as keys", () => {
+        return request(app)
+          .get("/api/topics").expect(200)
+          .then(({ body }) => {
+            console.log(body)
```
Comment (k-huggs) [inline __tests__/app.test.js]: delete console.log

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+beforeEach(() => {
+    return seed(data);
+});
+
+describe("GET /api/topics", () => {
+   describe("STATUS: 200", () => {
+    test("endpoint responds with a 200 status", () => {
+        return request(app).get("/api/topics").expect(200);
+    });
```
Comment (k-huggs) [inline __tests__/app.test.js]: you can combine these first two tests into one. As you test the status in the .expect(200) - so no need to have a separate test for it

File: controllers/controllers.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+const {selectTopics} = require('../models/models');
+
+exports.getTopics = (req, res, next) => {
+    selectTopics().then((allTopics) => {
+        res.status(200).send({allTopics})
+    })
```
Comment (k-huggs) [inline controllers/controllers.js]: .catch

**Comment by k-huggs** [comment]
couple amendments, then merge!

File: controllers/controllers.js (hunk 3 above)
Comment (solveigag) [reply controllers/controllers.js]: @k-huggs do i still need the .catch if my error won't be coming from model/controller? I thought this endpoint won't be triggered if the path is incorrect...

Comment (k-huggs) [reply controllers/controllers.js]: true, however, I would always put one on just incase something unusual came back

**Pull request merged**